torch/csrc/acc/Module.cpp

```diff
@@ -0,0 +1,205 @@
+  bool queryEvent(void* /*event*/) const override {
+    return true;
+  }
+  void destroyEvent(void* /*event*/, const c10::DeviceIndex /*device_index*/)
+      const noexcept override {}
+
+  // Stream-related functions
+  bool queryStream(const c10::Stream& /*stream*/) const override {
+    return true;
+  }
+  void synchronizeStream(const c10::Stream& /*stream*/) const override {}
+};
+
+namespace {
+
+static py::object g_hook;
+bool registerPythonPrivateUse1Hook(py::object hook) {
+  if (at::isPrivateUse1HooksRegistered()) {
+    return false;
+  }
+  g_hook = hook;
+  at::RegisterPrivateUse1HooksInterface(
+      hook.cast<PrivateUse1HooksInterface*>());
+  return true;
+}
+
+static py::object g_guard;
+bool registerPythonPrivateUse1DeviceGuard(py::object guard) {
+  if (c10::impl::hasDeviceGuardImpl(c10::DeviceType::PrivateUse1)) {
+    return false;
+  }
+  g_guard = guard;
+  c10::impl::registerDeviceGuard(
+      c10::DeviceType::PrivateUse1,
+      g_guard.cast<c10::impl::DeviceGuardImplInterface*>());
+  return true;
+}
+
+void cleanupPyObj() {
+  g_guard.release();
+  g_hook.release();
+}
+
+at::Tensor createEmptyTensor(
+    const std::vector<int64_t>& shape,
+    c10::ScalarType dtype) {
+  c10::Storage storage{
+      c10::Storage::use_byte_size_t{},
+      0,
+      c10::GetAllocator(c10::kMeta),
+      true,
+  };
+
+  c10::Device device(c10::DeviceType::PrivateUse1, 0);
+  storage.set_data_ptr_noswap(at::DataPtr{nullptr, device});
+  c10::DispatchKeySet key_set({c10::DispatchKey::PrivateUse1});
+  at::Tensor tensor = at::detail::make_tensor<at::TensorImpl>(
+      std::move(storage), key_set, c10::scalarTypeToTypeMeta(dtype));
+
+  std::vector<int64_t> strides(shape.size());
+  int64_t size = 1;
+  for (int64_t i = strides.size() - 1; i >= 0; --i) {
+    strides[i] = size;
+    size *= shape[i];
+  }
+
+  tensor.unsafeGetTensorImpl()->set_sizes_and_strides(shape, strides, 0);
+  return tensor;
+}
+} // namespace
+
+void initModule(PyObject* module) {
+  auto py_module = py::reinterpret_borrow<py::module>(module);
+
+  py::class_<at::PrivateUse1HooksInterface, PythonHooks>(
+      module, "PrivateUse1Hooks")
+      .def(py::init<>())
+      .def(
+          "has_primary_context",
+          &at::PrivateUse1HooksInterface::hasPrimaryContext)
+      .def("is_built", &at::PrivateUse1HooksInterface::isBuilt)
+      .def("is_available", &at::PrivateUse1HooksInterface::isAvailable);
+
+  py::class_<c10::impl::DeviceGuardImplInterface, PythonDeviceGuard>(
+      module, "DeviceGuard")
+      .def(py::init<>())
+      .def("type_", &c10::impl::DeviceGuardImplInterface::type);
+
+  py_module.def(
+      "register_python_privateuseone_hook", &registerPythonPrivateUse1Hook);
+  py_module.def(
+      "register_python_privateuseone_device_guard",
+      &registerPythonPrivateUse1DeviceGuard);
+  py_module.def("create_empty_tensor", &createEmptyTensor);
+
+  // Make sure to unclean up python objects when python interpreter exits
```

Comment:
Is there anything significant associated with these?  If not, I would just leak them.